db/languages.yml

```diff
@@ -108,6 +108,9 @@ python:
     python3.8:
       :name: "Python 3.8"
       :interpreter: /usr/bin/python3.8
+    python3.6pypy:
+      :name: "Python 3.6 (PyPy)"
```

Comment:
Indeed, pypy3.6 is a better name for internal purposes. However, PyPy might be confused as a language if it appears as `PyPy` in the dropdown. It is still the Python 3.6 language we are just running it by a less conventional method. Following what we used for JavaScript, I prefer the language to come first followed by version then interpreter if relevant. This also means PyPy appears in the "other" section which I think is better while we are experimenting with this

Reply:
Yet another idea: identifier "pypy3", name "Python 3.6 (PyPy 7.3)".

\- Reason for identifier: The package name only includes the major version (pypy3), so it's unlikely that we'll have multiple versions (e.g. pypy3.6 and pypy3.7). This also resolves the "[specific version](https://github.com/NZOI/nztrain/pull/104/files#r410231155)" comment below, in a way.

\- Reason for PyPy version in name: The PyPy version (7.3) doesn't map one-to-one to the Python version (e.g. PyPy 7.2 also supports Python 3.6.9). The [changelog](https://doc.pypy.org/en/latest/release-v7.3.0.html) doesn't look interesting, but the version may be useful to someone (especially since pypy's version isn't printed during submission, unlike the compiled languages). On the other hand, the "7.3" will need to be updated more frequently, so I'm OK with leaving it out.

\- Reason for "Python" in name: I think it's good to include "Python", because they *are* the same language and they are compatible; it will encourage people to try PyPy and get the speed improvement. Since PyPy will be in the "Other" section, I don't think there is risk of confusion (and it will probably always lag a couple of versions, so it should be relatively clear that "Python 3.8" is the current one and "Python 3.6 (PyPy 7.3)" is something older/weirder).

(Aside: I think it is more "correct" to have a separate LanguageGroup for PyPy, mainly because the current version of PyPy is independent from the current version of CPython. But that's really minor, probably not worth the effort.)

Reply:
Have gone with "Python 3.6 (PyPy 7.3)" since we probably aren't going to update the PyPy version unless reinstalling the server or explicitly updating PyPy to support a newer language version anyway